changelogs/current.yaml

```diff
@@ -20,6 +20,9 @@ minor_behavior_changes:
 - area: http2
   change: |
     Request authorities are now validated with a library function from QUICHE rather than nghttp2. This behavior change can be reverted by setting ``envoy.reloadable_features.http2_validate_authority_with_quiche`` to ``false``.
+- area: router redirect
+  change: |
+    when ``port_redirect`` is not specified, the request's port is used.
```

Comment:
Is this a breaking change? Our current API, which exposes this option, has the following enum option:

`FROM_PROTOCOL_DEFAULT: automatically set to 80 for HTTP and 443 for HTTPS.`

AFAIK this was the previous behavior in envoy with port unset. With this new change, it will use another port

Reply:
> AFAIK this was the previous behavior in envoy with port unset. With this new change, it will use another port

If port_redirect is set explicitly, there is no change in behavior. Does your API requires this param or can it be left unspecified?

Reply:
Sorry to clarify, when a user sets that in our API we do *not* set port_redirect and explicitly expect the "automatically set to 80 for HTTP and 443 for HTTPS.". This behavior is changed by this PR, which I think represents a breaking API change for envoy.

Reply:
Thanks for explaining. To be precise, Envoy did not add port 80 or 443, but simply stripped the incoming port. The result was that port was auto chosen based on the protocol: http or https. 
Solution to that problem would be to set `port_redirect` to zero for reusing the port from the request. The logic would be:
- port_redirect not specified - as before (strip port)
- port_redirect = 0 (reuse from request)
- port_redirect = value (use the value specified)

Reply:
With a `uint32` I don't think you can distinguish unset and 0?

Reply:
yeah.. you are right :-(

Reply:
API LGTM revoked until we resolve this, thanks for noticing @howardjohn

Reply:
@howardjohn if Istio's API says
```
FROM_PROTOCOL_DEFAULT: automatically set to 80 for HTTP and 443 for HTTPS
``` 
It could now explicitly set these published port values in
https://github.com/istio/istio/blob/96f6a4f16d5a5aa4905ff0c0f94d9dd607b5f5f3/pilot/pkg/networking/core/v1alpha3/route/route.go#L603
rather than implictly assuming envoy will

from the original API PR https://github.com/istio/api/pull/2088 it looks like this is what istio would have wanted envoy to do

Reply:
It really doesn't matter what Istio does or docs say, this is a breaking change in Envoy which will extend beyond Istio.